### ISSUE TITLE
Retain formatting during copy-paste

### ISSUE THREAD

**Issue by sundoulos2** [Automattic/wp-desktop#963]
#### Steps to reproduce
1. Format some text in an application (can be either native linux or Windows app running in WINE)
2. Copy the text to the clipboard
3. Paste the text into a new post

#### What I expected
The pasted text should have the same formatting as the copied text

#### What happened instead
All formatting was stripped off

#### App Version / OS version
wordpress.com-linux-x64-6.1.0
OpenSUSE Tumbleweed 2020-10-14
KDE Plasma 5.20.0

#### Screenshot / Video
![wp-deskotop screenshot](https://user-images.githubusercontent.com/26822854/96453799-fc6cb680-11e8-11eb-88d2-89c5851d9264.jpg)

#### Context / Source

Bi-directional copying works correctly between native linux applications (WPS Office, Libreoffice, etc), between various Windows apps running under WINE with each other and with linux apps. Pasting text into wp-desktop from either linux or WINE loses all formatting information except for bold and underline.
If I copy the text from any app to the online editor at HTMLEditor.in and then paste the source from there to wp-desktop code editor, I get the expected results.

**Comment by designsimply** [comment]
Thanks for the report and for including a screenshot! 

I believe the formatting behavior you are seeing may be expected. It's also worth noting that if the behavior you are describing happens the same way for the visual editor in a browser then we will want to move this issue upstream because in that case it is an issue with the editor directly and not with the WPDesktop app specifically.

I have a few questions:

If you use a browser and go to https://wordpress.com/post and paste content into the visual editor (not the HTML editor), does the same behavior happen in the browser that you're describing for the WPDesktop app? Note that the address should start with https://wordpress.com/post and if it does not then you may be using a different editor entirely.

> If I copy the text from any app to the online editor at HTMLEditor.in and then paste the source from there to wp-desktop code editor, I get the expected results.

Sorry, I'm finding this part a bit hard to follow. Are you pasting content from a rich editor app (such as WPS Office or Libreoffice) into a browser using an address starting with https://wordpress.com/post but using the HTML mode and then re-copying (while *still* in HTML mode) into the WPDesktop app and then switching to the visual editor and then you find the formatting looks exactly the same in the WPDesktop app as what we can see on the left in your screenshot?

**Comment by sundoulos2** [comment]
I do get the same results with the visual editor in a browser. (Note: a couple years ago, pasting formatted text from LibreOffice to the visual editor at wordpress.com retained the formatting. It quit working, maybe when wordpress switched to the new editor?)

As for your second question, I'm using the TinyMCE based editor hosted at http://htmleditor.in/index.html with the following steps:
1. Paste formatted text from a rich editor app into the WYSIWYG editor at http://htmleditor.in/index.html
2. Switch to Source view and copy the html source
3. Paste the html source into the WPDesktop app code editor
4. Exit the code editor and the formatting in the visual editor matches the original document.

I tried several WYSIWYG html editors and TinyMCE did the best job of converting rich text to html.

**Comment by designsimply** [comment]
Gotcha. Okay! And yes, there's a completely new editor being developed and it's a pretty big switch and is getting lots of development recently although it sounds like you prefer the way the previous editor worked for pasting text instead of how the newer editor is handling it. Based on my own experience in testing and triage, I think what I am seeing in your screenshots is the intended behavior. I would like to suggest asking about this directly in the repository where the editor is being developed https://github.com/WordPress/gutenberg/ because WordPress.com is using that editor.

I did a search https://github.com/WordPress/gutenberg/issues?q=is%3Aissue+in%3Atitle+paste looking for a similar issue but didn't spot anything open that looks exactly the same.

I propose that you submit an issue there with the following additional information:

* In the expected behavior section, mention specifics such as whether you expected single line breaks between headings and paragraphs to remain the same, you expected indented text (are yours using tabs or spaces?) to remain indented, and you expected colors and underlining to remain. Not all formatting was stripped in your example above, only some of it.
* Update your screenshot to show a browser view and not the WPDesktop—this is because the fix in this case needs to be done upstream.
* If possible, include a link to a sample Libreoffice document that can be downloaded and used for testing.
* Paste the same content into Google Docs and report whether the content pastes properly there (I noticed some developers working on the WordPress editor project asked this question in other similar issues).
* Leave out the part about the HTML editor to start, I think it potentially misdirects from the problem at hand—which is that you expect formatted text from a rich editor to match _exactly_ when pasted into the WordPress visual editor.
* Consider asking what the expected behavior is when pasting content from a rich editor such as Libreoffice into the WordPress editor. Is all formatting expected to be retained?

Let me know if you would like help re-submitting this issue, although I feel in this case it's best created by you as a user of the product and I'd like to help guide you in where to post and what questions to ask. 🙂

**Comment by sundoulos2** [comment]
Thanks for the suggestion. It will be a day or two before I can devote time to resubmitting it.

**Comment by sundoulos2** [comment]
Here 'tis:
[https://github.com/WordPress/gutenberg/issues/26434]

**Comment by nsakaimbo** [comment]
Thank you, @sundoulos2!

Closing this issue as it has been filed with the Gutenberg repository.
